api/internal/git/repospec_test.go

```diff
@@ -543,35 +560,11 @@ func TestNewRepoSpecFromUrl_Smoke(t *testing.T) {
 			repoSpec: RepoSpec{
 				Host:         "https://git@home.com/",
 				RepoPath:     "path/to/repository",
-				KustRootPath: "/path",
+				KustRootPath: "path",
 				Ref:          "branch",
 				GitSuffix:    ".git",
 			},
 		},
-		{
-			name:      "unsupported protocol after username (invalid but currently passed through to git)",
```

Comment:
This is a behaviour change vs before all our refactoring, though at the end of the day this URL could never work. This PR would require adding code to continue accepting this particular input, and since it isn't valid anyway, I didn't think it was worthwhile.